tezos/context/src/working_tree/mod.rs

```diff
@@ -77,6 +88,41 @@ pub enum Object {
     Commit(Box<Commit>),
 }
 
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, Hash, PartialEq, Eq)]
+pub struct ObjectReference {
```

Comment:
How about `KeyValueStoreBackend` having a type parameter? it is there that `ObjectReference` is used right? So the inmem implementation would have `HashId` and the ondisk implementation `AbsoluteOffset`. `TezedgeContext`, because it contains the repo inside will probably need to be parametrized over that type too.